### PR TITLE
executor: use single batch for insert

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
Use multi-batch for insert statement is dangerous.

### What is changed and how it works?
Batch the insert statement only when `tidb_batch_insert` is enabled.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

### CODE REVIEW

**Comment by jackysp** [comment]
The data race will be fixed in another PR.

File: executor/insert_common.go [hunk 1]
```diff
@@ -191,15 +191,13 @@ func (e *InsertValues) insertRows(ctx context.Context, exec func(ctx context.Con
 			return errors.Trace(err)
 		}
 		rows = append(rows, row)
-		if e.rowCount%uint64(batchSize) == 0 {
+		if batchInsert && e.rowCount%uint64(batchSize) == 0 {
```
Comment (crazycs520) [inline executor/insert_common.go]: What if `batchInsert ` is `false`?

File: executor/insert_common.go [hunk 2]
```diff
@@ -191,15 +191,13 @@ func (e *InsertValues) insertRows(ctx context.Context, exec func(ctx context.Con
 			return errors.Trace(err)
 		}
 		rows = append(rows, row)
```
Comment (crazycs520) [inline executor/insert_common.go]: What if `batchInsert ` is `false`?

File: executor/insert_common.go (hunk 1 above)
Comment (jackysp) [reply executor/insert_common.go]: If it is false, all the rows will be handled at 
https://github.com/pingcap/tidb/pull/9406/files#diff-0f31e77a0899889b749ff90c69acaaf6R204

**Comment by jackysp** [comment]
PTAL @crazycs520

**Comment by jackysp** [comment]
PTAL @crazycs520

**Pull request merged**